amazon-kinesis-client/src/test/java/software/amazon/kinesis/common/ConfigsBuilderTest.java

```diff
@@ -53,12 +54,27 @@ public class ConfigsBuilderTest {
     private static final String WORKER_IDENTIFIER = "worker-id";
 
     @Test
-    public void testTrackerConstruction() {
+    public void testTrackerConstructionByStreamName() {
         final String streamName = "single-stream";
         final ConfigsBuilder configByName = createConfig(streamName);
         final ConfigsBuilder configBySingleTracker = createConfig(new SingleStreamTracker(streamName));
 
-        for (final ConfigsBuilder cb : Arrays.asList(configByName, configBySingleTracker)) {
+        testTrackerConstruction(streamName, configBySingleTracker, configByName);
```

Comment:
This method tests multi-stream construction twice which is both redundant, and will have a higher blast-radius if multi-stream fails.

Consider splitting this into two test methods: single-stream and multi-stream. Single-stream may look something like:
```
for (final ConfigsBuilder cb : Arrays.asList(
    createConfig(streamName),
    createConfig(new SST(streamName)),
    createConfig(streamArn),
    createConfig(new SST(streamArn))) {
   // do the single-stream tests here
}
```